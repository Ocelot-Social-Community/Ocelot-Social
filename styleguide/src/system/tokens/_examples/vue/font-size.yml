#
# FONT-SIZE TOKENS
# Use these tokens to set a font-size.
#

props:
  - name: font-size-xxxx-large
    value: "3rem"
  - name: font-size-xxx-large
    value: "2.5rem"
  - name: font-size-xx-large
    value: "2rem"
  - name: font-size-x-large
    value: "1.5rem"
  - name: font-size-large
    value: "1.25rem"
  - name: font-size-base
    value: "1rem"
  - name: font-size-body
    value: "15px"
  - name: font-size-small
    value: "0.8rem"
  - name: font-size-x-small
    value: "0.7rem"
  - name: font-size-xx-small
    value: "0.6rem"
global:
  type: number
  category: font-size
