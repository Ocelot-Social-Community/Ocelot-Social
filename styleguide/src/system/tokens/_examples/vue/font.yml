#
# FONT TOKENS
# Use these tokens to set font-weight and font-family.
#

props:
  - name: font-family-heading
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-text
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-code
    value: "inconsolata, monospace"
    category: "font-family"

  - name: font-weight-regular
    value: "normal"
    category: "font-weight"
  - name: font-weight-bold
    value: "600"
    category: "font-weight"
global:
  type: ...
  category: font
