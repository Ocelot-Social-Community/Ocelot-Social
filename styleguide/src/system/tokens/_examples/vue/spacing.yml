#
# SPACING TOKENS
# Use these tokens to set padding, margin and position coordinates.
#

aliases:
  xxx-large:
    value: &xxx-large 120
  xx-large:
    value: &xx-large 60
  x-large:
    value: &x-large 40
  large:
    value: &large 28
  base:
    value: &base 20
  small:
    value: &small 12
  x-small:
    value: &x-small 6
  xx-small:
    value: &xx-small 4
  xxx-small:
    value: &xxx-small 2

props:
  - name: xxx-large
    value: *xxx-large
    category: space-size
  - name: xx-large
    value: *xx-large
    category: space-size
  - name: x-large
    value: *x-large
    category: space-size
  - name: large
    value: *large
    category: space-size
  - name: base
    value: *base
    category: space-size
  - name: small
    value: *small
    category: space-size
  - name: x-small
    value: *x-small
    category: space-size
  - name: xx-small
    value: *xx-small
    category: space-size
  - name: xxx-small
    value: *xxx-small
    category: space-size

  - name: space-xxx-large
    value: "{!xxx-large}px"
  - name: space-xx-large
    value: "{!xx-large}px"
  - name: space-x-large
    value: "{!x-large}px"
  - name: space-large
    value: "{!large}px"
  - name: space-base
    value: "{!base}px"
  - name: space-small
    value: "{!small}px"
  - name: space-x-small
    value: "{!x-small}px"
  - name: space-xx-small
    value: "{!xx-small}px"
  - name: space-xxx-small
    value: "{!xxx-small}px"

global:
  type: number
  category: space
