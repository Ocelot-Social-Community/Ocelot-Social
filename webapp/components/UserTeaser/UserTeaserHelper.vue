<template>
  <span v-if="!linkToProfile">
    <slot />
  </span>
  <button v-else-if="linkToProfile && showPopover && isTouchDevice" @click.prevent="openMenu">
    <slot />
  </button>
  <nuxt-link
    v-else
    :to="userLink"
    @mouseover.native="() => showPopover && openMenu()"
    @mouseleave.native="closeMenu"
  >
    <slot />
  </nuxt-link>
</template>

<script>
export default {
  name: 'UserTeaserHelper',
  props: {
    userLink: { type: Object },
    linkToProfile: { type: Boolean, default: true },
    showPopover: { type: Boolean, default: false },
    isTouchDevice: { type: Boolean, default: false },
  },
  methods: {
    openMenu() {
      this.$emit('open-menu')
    },
    closeMenu() {
      this.$emit('close-menu')
    },
  },
}
</script>
