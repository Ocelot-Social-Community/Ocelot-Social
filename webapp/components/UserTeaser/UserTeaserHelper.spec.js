import { render, waitFor, fireEvent } from '@testing-library/vue'
import { RouterLinkStub } from '@vue/test-utils'
import UserTeaserHelper from './UserTeaserHelper.vue'

const localVue = global.localVue

const userLink = {
  name: 'profile-id-slug',
  params: { slug: 'slug', id: 'id' },
}

let mockIsTouchDevice

jest.mock('../utils/isTouchDevice', () => ({
  isTouchDevice: jest.fn(() => mockIsTouchDevice),
}))

describe('UserTeaserHelper', () => {
  const Wrapper = ({
    withLinkToProfile = true,
    onTouchScreen = false,
    withPopoverEnabled = true,
    hoverDelay = 500,
  }) => {
    mockIsTouchDevice = onTouchScreen

    return render(UserTeaserHelper, {
      localVue,
      propsData: {
        userLink,
        linkToProfile: withLinkToProfile,
        showPopover: withPopoverEnabled,
        hoverDelay: hoverDelay,
      },
      stubs: {
        NuxtLink: RouterLinkStub,
      },
      slots: {
        default: '<div>Test Content</div>',
      },
    })
  }

  // Helper function for tests that need timers
  const withFakeTimers = (testFn) => {
    return async () => {
      jest.useFakeTimers()
      try {
        await testFn()
      } finally {
        jest.runOnlyPendingTimers()
        jest.useRealTimers()
      }
    }
  }

  describe('hover delay functionality', () => {
    it(
      'should emit open-menu after hover delay on desktop',
      withFakeTimers(async () => {
        const wrapper = Wrapper({
          withLinkToProfile: true,
          onTouchScreen: false,
          withPopoverEnabled: true,
          hoverDelay: 1000,
        })

        // Find the NuxtLink stub element
        const link = wrapper.container.firstChild
        expect(link).toBeTruthy()

        // Trigger mouseenter
        fireEvent.mouseEnter(link)

        // Menu should not be opened yet
        expect(wrapper.emitted()['open-menu']).toBeFalsy()

        // Advance time
        jest.advanceTimersByTime(1000)

        // Now open-menu should have been emitted
        await waitFor(() => {
          expect(wrapper.emitted()['open-menu']).toBeTruthy()
          expect(wrapper.emitted()['open-menu']).toHaveLength(1)
        })
      }),
    )

    it(
      'should not emit open-menu if mouse leaves before delay',
      withFakeTimers(async () => {
        const wrapper = Wrapper({
          withLinkToProfile: true,
          onTouchScreen: false,
          withPopoverEnabled: true,
          hoverDelay: 1000,
        })

        const link = wrapper.container.firstChild

        // Mouseenter + mouseleave
        fireEvent.mouseEnter(link)
        jest.advanceTimersByTime(500)
        fireEvent.mouseLeave(link)

        // Let the rest of the time pass
        jest.advanceTimersByTime(500)

        // open-menu should not have been emitted
        expect(wrapper.emitted()['open-menu']).toBeFalsy()
        // But close-menu should have been emitted
        expect(wrapper.emitted()['close-menu']).toBeTruthy()
        expect(wrapper.emitted()['close-menu']).toHaveLength(1)
      }),
    )

    it('should emit open-menu immediately on touch device button click', async () => {
      const wrapper = Wrapper({
        withLinkToProfile: true,
        onTouchScreen: true,
        withPopoverEnabled: true,
      })

      const button = wrapper.container.querySelector('button')
      expect(button).toBeTruthy()

      // Click on button
      fireEvent.click(button)

      // Should be emitted immediately, without timer
      expect(wrapper.emitted()['open-menu']).toBeTruthy()
      expect(wrapper.emitted()['open-menu']).toHaveLength(1)
    })
  })

  describe('with linkToProfile and popover enabled, on touch screen', () => {
    let wrapper
    beforeEach(() => {
      wrapper = Wrapper({ withLinkToProfile: true, onTouchScreen: true, withPopoverEnabled: true })
    })

    it('renders button', () => {
      expect(wrapper.container.querySelector('button')).toBeTruthy()
      expect(wrapper.container).toMatchSnapshot()
    })
  })

  describe('without linkToProfile', () => {
    let wrapper
    beforeEach(() => {
      wrapper = Wrapper({ withLinkToProfile: false, onTouchScreen: false })
    })

    it('renders span', () => {
      expect(wrapper.container.querySelector('span')).toBeTruthy()
      expect(wrapper.container).toMatchSnapshot()
    })
  })

  describe('with linkToProfile, on desktop', () => {
    let wrapper
    beforeEach(() => {
      wrapper = Wrapper({ withLinkToProfile: true, onTouchScreen: false })
    })

    it('renders link', () => {
      const routerLinkStub = wrapper.container.firstChild
      expect(routerLinkStub).toBeTruthy()
      expect(routerLinkStub.tagName).toBeTruthy()
      expect(wrapper.container).toMatchSnapshot()
    })
  })

  describe('timer cleanup', () => {
    it(
      'should clear timer on component unmount',
      withFakeTimers(async () => {
        const clearTimeoutSpy = jest.spyOn(global, 'clearTimeout')

        const wrapper = Wrapper({
          withLinkToProfile: true,
          onTouchScreen: false,
          withPopoverEnabled: true,
          hoverDelay: 5000,
        })

        // Start hover timer
        fireEvent.mouseEnter(wrapper.container.firstChild)

        // Unmount while timer is running
        wrapper.unmount()

        // clearTimeout should have been called
        expect(clearTimeoutSpy).toHaveBeenCalled()

        clearTimeoutSpy.mockRestore()
      }),
    )
  })

  describe('popover behavior without showPopover', () => {
    it('should not emit events when showPopover is false', () => {
      const wrapper = Wrapper({
        withLinkToProfile: true,
        onTouchScreen: false,
        withPopoverEnabled: false,
      })

      const link = wrapper.container.firstChild

      fireEvent.mouseEnter(link)
      fireEvent.mouseLeave(link)

      expect(wrapper.emitted()['open-menu']).toBeUndefined()
      expect(wrapper.emitted()['close-menu']).toBeUndefined()
    })
  })
})
