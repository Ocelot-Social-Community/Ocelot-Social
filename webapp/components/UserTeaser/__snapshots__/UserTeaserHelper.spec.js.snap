// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTeaserHelper with linkToProfile and popover enabled, on touch screen renders button 1`] = `
<div>
  <button />
</div>
`;

exports[`UserTeaserHelper with linkToProfile, on desktop renders link 1`] = `
<div>
  <a />
</div>
`;

exports[`UserTeaserHelper without linkToProfile renders span 1`] = `
<div>
  <span />
</div>
`;
